### PR TITLE
Add Duktape 2.1 results

### PR DESCRIPTION
**Pull request by svaarala**
- Add results for Duktape 2.1
- Move duktape engines in environments.json to fix their HTML layout grouping (they were grouped under "mobile" for some reason)

### CODE REVIEW

**Review by chicoxyzzy — APPROVED** [review]

File: data-es2016plus.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ exports.tests = [
           safaritp: true,
           webkit: true,
           duktape20: true,
+          duktape21: true,
```
Comment (afmenez) [inline data-es2016plus.js]: You don't need to add every version here.  What is true for `duktape20` is also true for `duktape20` and all following versions.

File: environments.json [hunk 2]
```diff
@@ -1500,6 +1500,20 @@
       "esintl"
     ]
   },
+  "duktape20": {
+    "full": "Duktape 2.0.x",
```
Comment (afmenez) [inline environments.json]: I believe you could drop de ".x", it is implicit in the other browsers listed here.

**Review by afmenez — CHANGES_REQUESTED** [review]
Removing the redundancies as described in the comments would simplify this patch a lot.

File: data-es2016plus.js (hunk 1 above)
Comment (svaarala) [reply data-es2016plus.js]: Ok, I'll make that change. But what metadata tells that duktape21 is a newer version of duktape20? There doesn't seem to be explicit metadata to that effect in environments.json. Or is it simply string sort based?

File: environments.json (hunk 2 above)
Comment (svaarala) [reply environments.json]: Ok, sounds good.

File: data-es2016plus.js (hunk 1 above)
Comment (afmenez) [reply data-es2016plus.js]: I believe this is the purpose of the "family": "Duktape" field.  And yes, I believe it is sort based.

Comment (svaarala) [reply data-es2016plus.js]: Right, but how does one know X is newer than Y in the same family? I guess it's based on ordering in environments.json, i.e. same family but later in the file is newer?

Comment (svaarala) [reply data-es2016plus.js]: In any case the results are correct with the proposed change, so I'll apply that change to all the data files and update the pull. Thanks!

**Comment by svaarala** [comment]
Suggested changes made, rebased and pushed.

**Comment by svaarala** [comment]
Lint failed due to a duplicate key; fixed that.

**Review by afmenez — APPROVED** [review]
LGTM

**Comment by svaarala** [comment]
The fix caused a HTML change and a Travis error from that. Hopefully both are now fixed :)

**Comment by svaarala** [comment]
Ok, everything seems to pass now.

**Review by afmenez — APPROVED** [review]
LGTM

**Pull request merged**